### PR TITLE
Login

### LINKED ISSUE #66: Sign In Page
# User Story
As a **user**,
I want to **sign in/up**,
so I can **have my own private lists**.

# Description

#Material
image

# Done

- [x] Handle password hashing
- [x] Implement UserIds and list ids
- [x] (...)

### PR DESCRIPTION
**Pull request by LebkuchenCJ**
Fix #66 

## Deployment:
https://pryces-login-aah3fymdycv7zmca9.herokuapp.com/

## Comments: 
 - You can now sign up and create your personal lists.
 - Only one email can be added
 - Only your own lists will be displayed (Filtering is yet frontend based as backend solution did not work)
 - Passwords will be encrypted with bcrypt

image

### CODE REVIEW

File: lib/routes/users.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+/* router.post("/login", async (request, response) => {
+  const { name, password } = response.body;
+  try {
+    const result = await checkUser(name, password);
+    response.status(200).json(result);
+  } catch (error) {
+    console.error(error);
+    response.status(400).send("Couldn't find user");
+  }
+}); */
```
Comment (manuelsanchez2) [inline lib/routes/users.js]: Do you need this later?

File: client/src/components/SignUpContainer.js [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+            <h2>Oh no!</h2>
+            <h3>This E-Mail already exists</h3>
+          </>
+        )}
+        {success && (
+          <>
+            <h2>Wohoo</h2>
+            <h3>You created a new account. Sign in and start saving.</h3>
+          </>
+        )}
```
Comment (manuelsanchez2) [inline client/src/components/SignUpContainer.js:29]: Love this!

File: client/src/components/LoginForm.js [hunk 3]
```diff
@@ -1,26 +1,54 @@
...
+            onChange={(event) => setEmail(event.target.value)}
+          />
         </div>
       </label>
       <label>
         Password
         <div>
-          <input placeholder="Enter your password" type="password" />
+          <input
+            placeholder="Enter your password"
```
Comment (manuelsanchez2) [inline client/src/components/LoginForm.js:42]: Sometimes you use English, sometimes you use German. Try to be consistent with only one of them

**Review by manuelsanchez2 — APPROVED** [review]
Really clean code! Like it! Keep this good job!

File: lib/routes/users.js (hunk 1 above)
Comment (LebkuchenCJ) [reply lib/routes/users.js]: Good point, I will delete it

File: client/src/components/LoginForm.js (hunk 3 above)
Comment (LebkuchenCJ) [reply client/src/components/LoginForm.js:42]: Will have a look, but I thought I will stick to it

**Pull request merged**